packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -234,16 +238,31 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
   constructor(scope: Construct, id: string, props: UserPoolClientProps) {
     super(scope, id);
 
+    this.oAuthFlows = props.oAuth?.flows ?? {
+      implicitCodeGrant: true,
+      authorizationCodeGrant: true,
+    };
+
+    let callbackUrls: string[] | undefined = props.oAuth?.callbackUrls;
+    if (this.oAuthFlows.authorizationCodeGrant || this.oAuthFlows.implicitCodeGrant) {
+      if (callbackUrls === undefined) {
+        callbackUrls = [ 'https://example.com' ];
+      } else if (callbackUrls.length === 0) {
+        throw new Error('callbackUrl must be specified when codeGrant or implicitGrant OAuth flows are enabled.');
```

Comment:
This means the user explicitly defined `[]` as the callback urls right? otherwise the default would apply.

```suggestion
throw new Error('callbackUrl must not be empty when codeGrant or implicitGrant OAuth flows are enabled.');
```